packages/jack_link/PKGBUILD

```diff
@@ -1,36 +1,38 @@
 # Maintainer: OSAMC <https://github.com/osam-cologne/archlinux-proaudio>
 # Contributor: Christopher Arndt <aur -at- chrisarndt -dot- de>
+# Contributor: Florian Hülsmann <fh@cbix.de>
 
-_link_commit="41d9aa111f702e78b6fbaee9d3e06dda1db6420d"
 pkgname=jack_link
-pkgver=0.2.3
+pkgver=0.2.4
 pkgrel=1
 pkgdesc='JACK transport bridge to Ableton Link'
 arch=(aarch64 x86_64)
 url='https://github.com/rncbc/jack_link'
 license=(GPL-2.0-or-later)
 depends=(gcc-libs glibc)
-makedepends=(asio jack)
+makedepends=(asio git jack)
 groups=(pro-audio)
-source=("$pkgname-$pkgver.tar.gz::https://github.com/rncbc/jack_link/archive/refs/tags/v$pkgver.tar.gz"
-        "link-$_link_commit.tar.gz::https://github.com/Ableton/link/archive/$_link_commit.tar.gz")
-sha256sums=('709947415eb4e3ac50d5aa55b69480d5e74b20f7fd8b6d61df33e129f40354fd'
-            '013a6fc8d24086904717e723667099cc6645e36f960be7046707e186e5626067')
+source=("git+https://github.com/rncbc/$pkgname#tag=v$pkgver"
+        'git+https://github.com/Ableton/link')
+sha256sums=('6f19378c1004f2512758c6f91bd5fad230526f6a11a5daabf3d33549749bdf9f'
+            'SKIP')
 
 prepare() {
-  cd $pkgname-$pkgver
-  test -d link && rm -rf link
-  ln -s "$srcdir"/link-$_link_commit link
+  cd $pkgname
+  # setup submodule
+  git submodule init link
+  git submodule set-url link "$srcdir"/link
+  git -c protocol.file.allow=always submodule update
```

Comment:
Correct me, if I'm wrong, but won't this compile against the current HEAD of the `link` repo, and not necessarily against the commit, that was current when the `jack_link` release was tagged? This would not be appropriate for a release package, imho, because it is not reproducible.

Reply:
No, the `submodule update` inside `prepare()` checks out the commit defined in the parent repo.
Also see https://wiki.archlinux.org/title/VCS_package_guidelines#Git_submodules for reference :)

Reply:
Ok, I got confused by looking only at the diff and assuming that the jack_link sources were still in tarball for.

Carry on, then :smiley: